module/VuFind/src/VuFind/Controller/EdsrecordController.php

```diff
@@ -80,7 +80,11 @@ protected function redirectToEbook($format, $method)
             }
             throw new ForbiddenException('Access denied.');
         }
-        return $this->redirect()->toUrl($driver->tryMethod($method));
+        $url = $driver->tryMethod($method);
+        if (!$url) {
+            throw new \VuFind\Exception\BadConfig('EDS API did not provide a URL for this record.');
```

Comment:
There is probably a better exception class to use for this.  I would argue it *is* caused by bad configuration, specifically on the EDS side, which is effectively what I'm saying in the exception message.  But the class may have been intended for only local config errors.

Reply:
You could also argue that it should not throw an exception at all, but should redirect to something more useful.  But I'm not sure what that would be and likely depends on the (unknown) context of why the URL lookup failed in the first place.